### PR TITLE
Few-shot MRC with Reasonbert example

### PR DESCRIPTION
**Pull request by bhavani105**
# Pull Request      

- README, script and evaluation

```Closes #(issue)```       

Notes:
- Replace `(issue)` **above** ↑↑↑ with the issue this PR closes to automatically link the two.
This must be done when the PR is created.
- Add multiple `Closes #(issue)` as needed.
- If this PR is work towards but does not close an issue, simply tag the issue without mentioning `Closes`.

## Description

**Describe** the changes proposed by this PR below to give the reviewer context below ↓↓↓

```(description)```     

### Request Review

Be sure to **request a review** from one or more reviewers (unless the PR is to an unprotected branch).

## Versioning

When opening a PR to make changes to PrimeQA (i.e. `primeqa/`) master, be sure to increment the version following
[semantic versioning](https://semver.org/).  The VERSION is stored [here](https://github.com/primeqa/primeqa/blob/main/VERSION) 
and is incremented using `bump2version {patch,minor,major}` as described in the (development guide documentation)[development.html].

 - [ ] Have you updated the VERSION?
 - [ ] Or does this PR not change the `primeqa` package or was not into master?

After pulling in changes from master to an existing PR, ensure the VERSION is updated appropriately.
This may require bumping the version again if it has been previously bumped.

If you're not quite ready yet to post a PR for review, feel free to open a [draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/).

### After Merging
If merging into master and VERSION was updated, after this PR is merged:
- [ ] [Create a release](https://docs.github.com/en/github/administering-a-repository/releasing-projects-on-github/managing-releases-in-a-repository#creating-a-release) from the master with version equal to [VERSION](https://github.com/primeqa/primeqa/blob/main/VERSION)
- [ ] Not merging to master or VERSION not updated

### Checklist

Review the following and mark as completed:
- [ ] [Tag an issue or issues](#what-does-this-pr-do) this PR addresses.
- [ ] Added [description](#description) of changes proposed.
- [ ] Review requested as [appropriate](#request-review).
- [ ] Version bumped as [appropriate](#Versioning).
- [ ] New classes, methods, and functions documented.
- [ ] Documentation for modified code is updated.
- [ ] Built documentation to confirm it renders as expected (see [here](development.md)).
- [ ] Code cleaned up and commented out code removed.
- [ ] Tests added to ensure all functionalities tested at >= 60% unit test coverage (see [here](development.md)).
- [ ] Code cleaned up and commented out code removed.
- [ ] Release created [as needed](#after-merging) after merging.

### CODE REVIEW

**Review by avisil — APPROVED** [review]
LGTM! Thanks @xiang-deng and @bhavani105 ! Looking to get this merged.

**Comment by xiang-deng** [comment]
@bhavani105 Thanks for helping with the integration!

**Review by riyazbhat — APPROVED** [review]

File: examples/reasonbert/README.md [hunk 1]
```diff
@@ -0,0 +1,220 @@
+# Few Shot Fine-tuning with ReasonBERT Models
+
+This README provides instructions on how to fine-tune a extractive reader starting with [ReasonBERT](https://huggingface.co/osunlp/ReasonBERT-RoBERTa-base) pre-trained model in a few-shot setting.
```
Comment (seirasto) [inline examples/reasonbert/README.md:3]:
```suggestion
This README provides instructions on how to fine-tune an extractive reader starting with [ReasonBERT](https://huggingface.co/osunlp/ReasonBERT-RoBERTa-base) pre-trained model in a few-shot setting.  
```

File: examples/reasonbert/README.md [hunk 2]
```diff
@@ -0,0 +1,220 @@
+# Few Shot Fine-tuning with ReasonBERT Models
+
+This README provides instructions on how to fine-tune a extractive reader starting with [ReasonBERT](https://huggingface.co/osunlp/ReasonBERT-RoBERTa-base) pre-trained model in a few-shot setting.  
+
+The datasets used for evaluation are subsets of the [MRQA](https://github.com/mrqa/MRQA-Shared-Task-2019) dataset specifically:
```
Comment (seirasto) [inline examples/reasonbert/README.md:5]:
```suggestion
The datasets used for evaluation are subsets of the [MRQA](https://github.com/mrqa/MRQA-Shared-Task-2019) dataset. Specifically:
```

File: examples/reasonbert/README.md [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+
+    - SQuAD
+    - NaturalQuestionsShort
+    - NewsQA 
+    - TriviaQA-web
+    - SearchQA
+
+## What is ReasonBERT
+
+ReasonBERT is a pretraining method that augments language models with the ability to reason over long-range relations and multiple contexts as described in [ReasonBERT: Pre-trained to Reason with Distant Supervision](https://arxiv.org/pdf/2109.04912).  As detailed in the paper, ReasonBERT proposes a distant supervision method to create pretrainig examples that require long range reasoning.  It pairs a query sentence with multiple relevant pieces of evidence drawn from possibly different places and defines a new LM pre-training objective, span reasoning, to recover entity spans that are masked out from the query sentence by jointly reasoning over the query sentence and the relevant evidence.
```
Comment (seirasto) [inline examples/reasonbert/README.md:15]:
```suggestion
ReasonBERT is a pretraining method that augments language models with the ability to reason over long-range relations and multiple contexts as described in [ReasonBERT: Pre-trained to Reason with Distant Supervision](https://arxiv.org/pdf/2109.04912).  As detailed in the paper, ReasonBERT proposes a distant supervision method to create pre-training examples that require long range reasoning.  It pairs a query sentence with multiple relevant pieces of evidence drawn from possibly different places and defines a new LM pre-training objective, span reasoning, to recover entity spans that are masked out from the query sentence by jointly reasoning over the query sentence and the relevant evidence. 
```

What is different "places"? Does that mean contexts?

File: examples/reasonbert/README.md [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+
+ReasonBERT is a pretraining method that augments language models with the ability to reason over long-range relations and multiple contexts as described in [ReasonBERT: Pre-trained to Reason with Distant Supervision](https://arxiv.org/pdf/2109.04912).  As detailed in the paper, ReasonBERT proposes a distant supervision method to create pretrainig examples that require long range reasoning.  It pairs a query sentence with multiple relevant pieces of evidence drawn from possibly different places and defines a new LM pre-training objective, span reasoning, to recover entity spans that are masked out from the query sentence by jointly reasoning over the query sentence and the relevant evidence. 
+
+Experiments on a variety of extractive question answering datasets show that ReasonBERT achieves strong performance compared to a variety of baselines.
+
+Under the few-shot setting, ReasonBERT substantially outperforms a RoBERTa baseline on the extractive question answering task. 
+
+Here we show how to replicate the results on the extractive question answering datasets using the [PrimeQA reader](../../primeqa/mrc/README.md). See section [Evalustion Scores](#evaluation-scores)
```
Comment (seirasto) [inline examples/reasonbert/README.md:21]:
```suggestion
Here we show how to replicate the results on the extractive question answering datasets using the [PrimeQA reader](../../primeqa/mrc/README.md). See section [Evaluation Scores](#evaluation-scores)
```

**Review by seirasto — APPROVED** [review]
LGTM. I've added some suggestions for the README (mostly typos), please take a look.

**Pull request merged**